google/cloud/storage/blob.py

```diff
@@ -943,7 +937,25 @@ def _do_download(
             downloads where chunk_size is set) an INFO-level log will be
             emitted. Supported values are "md5", "crc32c" and None. The default
             is "md5".
+
+        :type retry: google.api_core.retry.Retry
+        :param retry: (Optional) How to retry the RPC. A None value will disable
+            retries. A google.api_core.retry.Retry value will enable retries,
+            and the object will configure backoff and timeout options. Custom
+            predicates (customizable error codes) are not supported for media
+            operations such as this one.
+
+            This private method does not accept ConditionalRetryPolicy values
+            because the information necessary to evaluate the policy is instead
+            evaluated in client.download_blob_to_file().
+
+            See the retry.py source code and docstrings in this package
+            (google.cloud.storage.retry) for information on retry types and how
+            to configure them.
         """
+
+        retry_strategy = _api_core_retry_to_resumable_media_retry(retry)
```

Comment:
Would it be reasonable to expose a configurable option for mutating which errors are retried in ResumableMedia package or a pass in a lambda?

Reply:
I did think about that, but because both error codes and exceptions are covered, they're covered in slightly different ways and exceptions are mutated through the libraries we're using in the transport layer, I felt it was not practical for this PR. We would need to think about a more comprehensive unification of code or else an abstraction of the exceptions and error codes for that to work.